Swift/BRCrypto/BRCryptoWalletManager.swift

```diff
@@ -401,7 +391,9 @@ extension WalletManager {
                                                           event: WalletManagerEvent.syncEnded(error: nil))
 
                 case CRYPTO_WALLET_MANAGER_EVENT_BLOCK_HEIGHT_UPDATED:
-                    manager.height = event.u.blockHeight.value
+                    manager.listener?.handleManagerEvent (system: manager.system,
+                                                          manager: manager,
+                                                          event: WalletManagerEvent.blockUpdated(height: event.u.blockHeight.value))
```

Comment:
Is this the place to handle the TransferEvent.confirmation event?

Reply:
Confirmation is reflected in the transfer's state. If that's not what you had intended, I would think any event generation would be at the crypto C layer.

Reply:
I don't think confirmations is reflected in the transfer state because confirmations is a combination of transfer state and network block height.  

The original idea was for an App transfer view needing to update the display as blocks arrive and confirmations change.  The view could listen on Transfer events alone (if TransferEvent.confirmations exists) or on Transfer events (for other updates - confirmed, etc) and WalletManager.blockUpdated  events.  I thought listening just on Transfer events when displaying a TransferView was more natural.  Is the Java CryptoDemo displaying 'confirmations'?

Reply:
The Java has the same behaviour as the Swift algorithm (shown below).

From Swift/BRCrypto/BRCryptoTransfer.swift
```
    /// An optional confirmation.
    public var confirmation: TransferConfirmation? {
        if case .included (let confirmation) = state { return confirmation }
        else { return nil }
    }

    public func confirmationsAt (blockHeight: UInt64) -> UInt64? {
        return confirmation
            .flatMap { blockHeight >= $0.blockNumber ? (1 + blockHeight - $0.blockNumber) : nil }
    }

    /// The confirmations of transfer at the current network `height`.
    public var confirmations: UInt64? {
        return confirmationsAt (blockHeight: wallet.manager.network.height)
    }
```